ophyd/status.py

```diff
@@ -264,12 +264,35 @@ def __init__(self, positioner, target, *, start_ts=None,
         self.pos = positioner
         self.target = target
         self.start_ts = start_ts
+        self.start_pos = self.pos.position
         self.finish_ts = None
         self.finish_pos = None
 
+        self._watchers = []
+
         # call the base class
         super().__init__(positioner, **kwargs)
 
+    def watch(self, func):
+        """
+        Subscribe to notifications about progress. Useful for progress bars.
+
+        Parameters
+        ----------
+        func : callable
+            Expected signature: ``func(status_obj, fraction_done)``
+        """
+        self._watchers.append(func)
+
+    def notify_watchers(self):
+        try:
+            fraction = ((self.pos.position - self.start_pos) /
+                        (self.target - self.start_pos))
+        except Exception as exc:
+            fraction = exc
```

Comment:
It makes sense, just surprising.  Maybe just print the traceback and don't call the callback?